### PR TITLE
Address Sphinx 4.0 upgrade

### PR DESCRIPTION
**Pull request by bchen1116**
Sphinx v4.0.0 removes support for `add_javascript` and `add_stylesheet` and replaces them with other calls. 

We cap the version to <4.0.0 to avoid issues with [cross-referencing](https://readthedocs.com/projects/feature-labs-inc-evalml/builds/629821/).  (depreciated API docs [here](https://www.sphinx-doc.org/ar/master/extdev/deprecated.html?highlight=add_javascript#deprecated-apis))

### CODE REVIEW

**Comment by bchen1116** [comment]
New issue to unpin Sphinx [here](https://github.com/alteryx/evalml/issues/2246).

This PR will just cap Sphinx version <4.0.0 for now.

**Review by jeremyliweishih — APPROVED** [review]
looks good! Might need @chukarsten or @dsherry to merge in the dependency PR first so the dependency test passes here.

**Review by freddyaboulton — APPROVED** [review]
Thanks @bchen1116 !

**Review by angela97lin — APPROVED** [review]
LGTM!

Also filed https://github.com/alteryx/evalml/issues/2248 so we can automatically catch stuff like this in the future :)

**Review by ParthivNaresh — APPROVED** [review]

**Review by freddyaboulton — APPROVED** [review]
Thanks so much @bchen1116 !

File: evalml/tests/utils_tests/test_dependencies.py [hunk 1]
```diff
@@ -5,12 +5,20 @@
 import requirements
 
 
+def _get_req_name(name):
+    if name == 'imbalanced-learn':
+        return 'imblearn'
+    elif name == 'MarkupSafe':
+        return 'markupsafe'
```
Comment (dsherry) [inline evalml/tests/utils_tests/test_dependencies.py:12]: Damn that's tough. Nice patch though.

File: requirements.txt [hunk 2]
```diff
@@ -5,6 +5,7 @@ ipywidgets>=7.5
 xgboost>=0.82,<1.3.0
 catboost>=0.20
 lightgbm>=2.3.1,<3.1.0
+MarkupSafe==1.1.1
```
Comment (dsherry) [inline requirements.txt:8]: @bchen1116 why add this to our requirements? Shouldn't this only be in our docs requirements?

**Review by dsherry — APPROVED** [review]
@bchen1116 thank you!

I approved so that we can unblock `main` builds. But I do want to understand why we have to add `MarkupSafe` to `requirements.txt`. I thought its only required to build the docs, not required to run evalml.

File: requirements.txt (hunk 2 above)
Comment (bchen1116) [reply requirements.txt:8]: When I just added this to the doc requirements, we still had failing tests. I'm not sure why it happened, but adding it to requirements solved the issue

Comment (freddyaboulton) [reply requirements.txt:8]: RTD installs the docs requirements then evalml. evalml installs `MarkUpSafe` via `ipywidgets` so if we pin markupsafe in docs requirements, it'll still be overwritten when RTD installs evalml

Comment (dsherry) [reply requirements.txt:8]: @freddyaboulton understood

[Here](https://github.com/alteryx/evalml/blob/main/.readthedocs.yaml#L21) is where we configure what gets installed in our RTD builds. I wonder if we now need another requirements file, which gets installed _after_ evalml is installed, and we can move the markupsafe pin there. I'm trying that out now in a draft PR #2261

**Pull request merged**